add warning about use directive on custom components

tennox opened the pull request:
As prompted here: 
https://github.com/solidjs/solid/discussions/722#discussioncomment-1646016

<!-- Thank you for taking the time to open this PR! We appreciate your contribution and effort in helping improve the project. -->

- [x] I have read the [Contribution guide](https://github.com/solidjs/solid-docs/blob/main/CONTRIBUTING.md)
- [x] This PR references an issue (except for typos, broken links, or other minor problems)


### Related issues & labels

- https://github.com/solidjs/solid/discussions/722

tennox committed:
add warning about use directive on custom components

As prompted here
https://github.com/solidjs/solid/discussions/722#discussioncomment-1646016

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

titoBouzout commented on src/routes/reference/jsx-attributes/use.mdx:
````diff
@@ -34,3 +34,7 @@ declare module "solid-js" {
 ```
+
+<Callout type="caution" title="Custom components">
+This directive does not currently work on custom components. [details here](https://github.com/solidjs/solid/discussions/722)
````
> Directives only work with native HTML elements (HTML/SVG/MathML/Custom Elements). Directives are not forwarded and wont work in user defined components, such `<TextInput use:myinput={[..]}/>` .  
> [See also](https://github.com/solidjs/solid/discussions/722).

I reversed the explanation, because its more clear that way. At first, I thought it was saying that doesn't work with custom elements. I'd also change the title of the callout to "Limitations"

Thanks!

tennox replied:
done - https://github.com/solidjs/solid-docs/pull/952/commits/d9d2ae680b20dac96fd6303f11863975b22f8013 & https://github.com/solidjs/solid-docs/pull/952/commits/a839fe5912e548d1e57c15e6d7b7e64000036e74

titoBouzout replied:
Thanks!

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

kodiakhq[bot] committed:
Merge branch 'main' into patch-1

tennox committed:
change wording of custom components warning

as suggested here https://github.com/solidjs/solid-docs/pull/952#pullrequestreview-2458910775

tennox committed:
change title to Limitations

titoBouzout commented:
@atilafassina this PR looks good to me, thanks!
